### PR TITLE
Support for execution module status on AIX

### PR DESCRIPTION
**Pull request by dmurphy18**
### What does this PR do?
Provides for support of status execution module on AIX

### What issues does this PR fix or reference?
https://github.com/saltstack/zh/issues/1206

### Previous Behavior
Remove this section if not relevant
Either returned unsupported platform, or traceback of invalid command

### New Behavior
Provides full support for the status execution module status on AIX, except for custom which requires further support of grains on AIX.

### Tests written?
No - unit tested by hand

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

File: salt/modules/status.py [hunk 1]
```diff
@@ -62,10 +66,57 @@ def _number(text):
...
+    t=$(LC_ALL=POSIX ps -o etime= -p 1)
+    d=0 h=0
+    case $t in *-*) d=${t%%-*}; t=${t#*-};; esac
+    case $t in *:*:*) h=${t%%:*}; t=${t#*:};; esac
+    s=$((d*86400 + h*3600 + ${t%%:*}*60 + ${t#*:}))
+
+    t is 7-20:46:46
+    '''
+    boot_secs = 0
+    res = __salt__['cmd.run_all']('ps -o etime= -p 1')
```
Comment (s0undt3ch) [inline salt/modules/status.py]: Shouldn't we be passing arguments as a list instead of a string?

File: salt/modules/status.py [hunk 2]
```diff
@@ -62,10 +66,57 @@ def _number(text):
...
+    boot_secs = _number(days[0]) * 86400 + _number(hms[0]) * 3600 + _number(hms[1]) * 60 + _number(hms[2])
+    return boot_secs
+
+
+def _aix_loadavg():
+    '''
+    Return the load average on AIX
+    '''
+    #  03:42PM   up 9 days,  20:41,  2 users,  load average: 0.28, 0.47, 0.69
+    uptime = __salt__['cmd.run']('uptime')
```
Comment (s0undt3ch) [inline salt/modules/status.py]: Same as above

File: salt/modules/status.py [hunk 3]
```diff
@@ -62,10 +66,57 @@ def _number(text):
...
+    return {'1-min': load_avg[1].strip(','),
+            '5-min': load_avg[2].strip(','),
+            '15-min': load_avg[3]}
+
+
+def _aix_nproc():
+    '''
+    Return the maximun number of PROCESSES allowed per user on AIX
+    '''
+    nprocs = __salt__['cmd.run']('lsattr -E -l sys0 | grep maxuproc', python_shell=True).split()
```
Comment (s0undt3ch) [inline salt/modules/status.py]: Same as above

File: salt/modules/status.py [hunk 4]
```diff
@@ -304,11 +369,45 @@ def sunos_cpustats():
...
 
+    def aix_cpustats():
+        '''
+        AIX specific implementation of cpustats
+        '''
+        ret = {}
+        ret['mpstat'] = []
+        procn = None
+        fields = []
+        for line in __salt__['cmd.run']('mpstat -a').splitlines():
```
Comment (s0undt3ch) [inline salt/modules/status.py]: Same as above

File: salt/modules/status.py [hunk 5]
```diff
@@ -366,10 +468,91 @@ def freebsd_meminfo():
...
+        '''
+        AIX specific implementation of meminfo
+        '''
+        ret = {}
+        ret['svmon'] = []
+        ret['vmstat'] = []
+        procn = None
+        fields = []
+        pagesize_flag = False
+        for line in __salt__['cmd.run']('svmon -G').splitlines():
```
Comment (s0undt3ch) [inline salt/modules/status.py]: Same as above

File: salt/modules/status.py [hunk 6]
```diff
@@ -366,10 +468,91 @@ def freebsd_meminfo():
...
+                procn = len(ret['svmon'])
+                ret['svmon'].append({})
+                comps = line.split()
+                ret['svmon'][procn][comps[0]] = {}
+                for i in range(0, len(fields)):
+                    if len(comps) > i:
+                        ret['svmon'][procn][comps[0]][fields[i]] = comps[i]
+                continue
+
+        for line in __salt__['cmd.run']('vmstat -v').splitlines():
```
Comment (s0undt3ch) [inline salt/modules/status.py]: Same as above

File: salt/modules/status.py [hunk 7]
```diff
@@ -490,12 +676,82 @@ def sunos_cpuinfo():
...
 
+    def aix_cpuinfo():
+        '''
+        AIX  specific cpuinfo implementation
+        '''
+        ret = {}
+        ret['prtconf'] = []
+        ret['lparstat'] = []
+        procn = None
+        for line in __salt__['cmd.run']('prtconf | grep -i "Processor"', python_shell=True).splitlines():
```
Comment (s0undt3ch) [inline salt/modules/status.py]: Not here since you're requiring a shell because of the pipe.

File: salt/modules/status.py [hunk 8]
```diff
@@ -561,11 +820,67 @@ def generic_diskstats():
...
+    def aix_diskstats():
+        '''
+        AIX specific implementation of diskstats
+        '''
+        ret = {}
+        procn = None
+        fields = []
+        disk_name = ''
+        disk_mode = ''
+        for line in __salt__['cmd.run']('iostat -dDV').splitlines():
```
Comment (s0undt3ch) [inline salt/modules/status.py]: Arguments as a list of strings?

File: salt/modules/status.py [hunk 9]
```diff
@@ -793,11 +1122,41 @@ def sunos_netstats():
...
 
+    def aix_netstats():
+        '''
+        AIX specific netstats implementation
+        '''
+        ret = {}
+        fields = []
+        procn = None
+        proto_name = None
+        for line in __salt__['cmd.run']('netstat -s').splitlines():
```
Comment (s0undt3ch) [inline salt/modules/status.py]: Same as above.

File: salt/modules/status.py [hunk 10]
```diff
@@ -906,11 +1268,75 @@ def sunos_netdev():
...
+            #root@la68pp002_pub:/opt/salt/lib/python2.7/site-packages/salt/modules# netstat -i -n -I en0 -f inet6
+            #Name  Mtu   Network     Address            Ipkts Ierrs    Opkts Oerrs  Coll
+            #en0   1500  link#3      e2.eb.32.42.84.c 10029668     0   446490     0     0
+            #en0   1500  172.29.128  172.29.149.95    10029668     0   446490     0     0
+            #root@la68pp002_pub:/opt/salt/lib/python2.7/site-packages/salt/modules# netstat -i -n -I en0 -f inet6
+            #Name  Mtu   Network     Address            Ipkts Ierrs    Opkts Oerrs  Coll
+            #en0   1500  link#3      e2.eb.32.42.84.c 10029731     0   446499     0     0
+
+            netstat_ipv4 = __salt__['cmd.run']('netstat -i -n -I {dev} -f inet'.format(dev=dev)).splitlines()
+            netstat_ipv6 = __salt__['cmd.run']('netstat -i -n -I {dev} -f inet6'.format(dev=dev)).splitlines()
```
Comment (s0undt3ch) [inline salt/modules/status.py]: Same as above

File: salt/modules/status.py [hunk 11]
```diff
@@ -1021,6 +1453,7 @@ def linux_version():
     get_version = {
         'Linux': linux_version,
         'FreeBSD': lambda: __salt__['cmd.run']('sysctl -n kern.version'),
+        'AIX': lambda: __salt__['cmd.run']('oslevel -s'),
```
Comment (s0undt3ch) [inline salt/modules/status.py]: Same as above

**Review by s0undt3ch — COMMENTED** [review]
See my above comments unless we're splitting the command string if python_shell is not true

File: salt/modules/status.py (hunk 1 above)
Comment (dmurphy18) [inline salt/modules/status.py]: @s0undt3ch following the style that is already existing in the file, for example
line 432, 444, 555 etc.

The use of cmd.run with a string such as 'iostat -xzd' is rampant in the file.

File: salt/modules/status.py (hunk 2 above)
Comment (dmurphy18) [inline salt/modules/status.py]: see comment above

File: salt/modules/status.py (hunk 3 above)
Comment (dmurphy18) [inline salt/modules/status.py]: see comment above

File: salt/modules/status.py (hunk 4 above)
Comment (dmurphy18) [inline salt/modules/status.py]: see comment above

File: salt/modules/status.py (hunk 5 above)
Comment (dmurphy18) [inline salt/modules/status.py]: see comment above

File: salt/modules/status.py (hunk 6 above)
Comment (dmurphy18) [inline salt/modules/status.py]: see comment above

File: salt/modules/status.py (hunk 8 above)
Comment (dmurphy18) [inline salt/modules/status.py]: see comment above

File: salt/modules/status.py (hunk 10 above)
Comment (dmurphy18) [inline salt/modules/status.py]: see comment above

File: salt/modules/status.py (hunk 11 above)
Comment (dmurphy18) [inline salt/modules/status.py]: see comment above

**Review by s0undt3ch — APPROVED** [review]
Ignore my comments above. If the shell command is not passed as a list, the `cmdmod` module will split it up into a list of strings if `python_shell=False` which is exactly what I was suggesting.
Since `cmdmod` will do that for you, there's no other reason in this code suggesting that you should not rely on that.
Basically, top notch work!

**Pull request merged**